lib/_http_outgoing.js

```diff
@@ -534,6 +552,9 @@ function processHeader(self, state, key, value, validate) {
 function storeHeader(self, state, key, value, validate) {
   if (validate)
     validateHeaderValue(key, value);
+  if (StringPrototypeToLowerCase(key) === 'content-length') {
+    self[kContentLength] = value;
```

Comment:
It would be more efficient to set this property in `matchHeader`, which already checks for this header.

Reply:
Understood, will do that also @mscdex can you please tell me when I should rebase? Should I first make the changes, fix markdown linting and then clean up the git history or do you have something else in mind?

Reply:
That's totally up to you. You can just make additional commits and then rebase once everything is done and all tests are passing or you can continually rebase. My personal preference is the latter because I like to keep my PRs in a landable state (even though whoever lands the changes will usually squash any unnecessary commits anyway), but to each their own.